src/commands/download_file.ts

```diff
@@ -0,0 +1,36 @@
+import { EID } from 'ardrive-core-js';
+import { resolve as resolvePath } from 'path';
+import { cliArDriveAnonymousFactory, cliArDriveFactory } from '../index';
+import { CLICommand, ParametersHelper } from '../CLICommand';
+import { CLIAction } from '../CLICommand/action';
+import {
+	DriveIdParameter,
+	DrivePrivacyParameters,
+	FileIdParameter,
+	LocalFilePathParameter
+} from '../parameter_declarations';
+
+new CLICommand({
+	name: 'download-file',
+	parameters: [FileIdParameter, LocalFilePathParameter, DriveIdParameter, ...DrivePrivacyParameters],
+	action: new CLIAction(async (options) => {
+		const parameters = new ParametersHelper(options);
+		const fileId = parameters.getRequiredParameterValue(FileIdParameter, EID);
+		const destFolderPath = resolvePath(parameters.getParameterValue(LocalFilePathParameter) || './');
+
+		if (await parameters.getIsPrivate()) {
+			const driveId = parameters.getRequiredParameterValue(DriveIdParameter);
```

Comment:
What if I provide the drive key as parameter? Drive ID shouldn't be required at all

Reply:
We should probably just eat the `getDriveIdForFileId` GQL call rather than having a driveID param. That is the pattern that most of the other commands are using.

You do bring up a great point here in that case too though, we wouldn't even need the drive ID if they supplied a driveKey. That inefficiency shows up in other commands where we don't pass down a driveId too, like `file-info` for example.

Reply:
I agree. Let's make driveKey and driveID optional for this command if possible.

Reply:
This was already applied :+1: